packages/common/Semaphore.js

```diff
@@ -0,0 +1,133 @@
+const {
+  isConditionalCheckException,
+  ResourcesLockedError
+} = require('./errors');
+const log = require('./log');
+
+class Semaphore {
+  constructor(docClient, tableName) {
+    this.docClient = docClient;
+    this.tableName = tableName;
+  }
+
+  async create(key) {
+    try {
+      const params = {
+        TableName: this.tableName,
+        Item: {
+          key,
+          semvalue: 0
+        },
+        ConditionExpression: '#key <> :key',
+        ExpressionAttributeNames: { '#key': 'key' },
+        ExpressionAttributeValues: { ':key': key }
+      };
+      await this.docClient.put(params).promise();
+    } catch (error) {
+      // Only re-throw errors that are not conditional check failures. A
+      // conditional check failure here means that a row tracking the semaphore
+      // for this key already exists, which is expected after the first operation.
+      if (!isConditionalCheckException(error)) {
+        throw error;
+      }
+    }
+  }
+
+  get(key) {
```

Comment:
Could this return `response.Item`?

Reply:
I definitely could, but to do it right I feel like I would basically be copying https://github.com/nasa/cumulus/blob/master/packages/api/models/base.js#L221, which feels not ideal.

Suggestions? This is kind of what I was getting at before in that several of the methods on this class are what we have already for our models. We can't extend this class on our models or anything though because the common package doesn't depend on the API package, nor should it.